github/commitstatus.go

```diff
@@ -175,9 +181,29 @@ func httpRequestDo(req *http.Request) (httpResponse, error) {
 	response.oauthInfo = fmt.Sprintf("X-Accepted-Oauth-Scopes: %v, X-Oauth-Scopes: %v",
 		resp.Header.Get("X-Accepted-Oauth-Scopes"), resp.Header.Get("X-Oauth-Scopes"))
 
-	response.rateLimitRemaining, _ = strconv.Atoi(resp.Header.Get("x-ratelimit-remaining"))
-	rateLimitReset, _ := strconv.Atoi(resp.Header.Get("x-ratelimit-reset"))
-	response.rateLimitReset = time.Unix(int64(rateLimitReset), 0)
+	if limit := resp.Header.Get("X-RateLimit-Remaining"); limit != "" {
+		v, err := strconv.Atoi(limit)
```

Comment:
`v` is really too short also if this is Go, `val` is OK.
I find the "failed to convert" confusing: as a user (so not seeing the code), what does it mean?
It is enough to say: 
```go
fmt.Errorf("X-RateLimit-Remaining: %s", err)
```
which would print:
```
X-RateLimit-Remaining: strconv.Atoi: parsing "foo": invalid syntax
```
As proof, see https://go.dev/play/p/6K64il-jIE8

Reply:
Consider also that strconv.Atoi is documented to return 0 in case of error, so we could simply write
```go
if limit := resp.Header.Get("X-RateLimit-Remaining"); limit != "" {
		response.rateLimitRemaining, err = strconv.Atoi(limit)
		if err != nil {
			return httpResponse{}, fmt.Errorf("X-RateLimit-Remaining: %s", err)
		}
}
```
or do as 1 line:
```
    // strconv.Atoi returns 0 in case of error.
	response.rateLimitRemaining, _ = strconv.Atoi(resp.Header.Get("X-RateLimit-Remaining"))
```

in the 3 cases, I think the result will be the same, AS LONG as this function is called on `response` only once. If it is called multiple times, then we must pick the correct behavior.

I don't have a strong opinion, as long as the code is consistent.

Reply:
> Consider also that strconv.Atoi is documented to return 0 in case of error

but not always :)

```
val, err := strconv.Atoi("9223372036854775808")
err = fmt.Errorf("X-RateLimit-Remaining: %s", err)
fmt.Println(val, err)
```
results in:
`9223372036854775807 X-RateLimit-Remaining: strconv.Atoi: parsing "9223372036854775808": value out of range`

Reply:
The example I gave was in case the header key was missing, which would result in a 0.

This code:
```go
if limit := resp.Header.Get("X-RateLimit-Remaining"); limit != "" {
    response.rateLimitRemaining, err = strconv.Atoi(limit)
    if err != nil {
        return httpResponse{}, fmt.Errorf("X-RateLimit-Remaining: %s", err)
    }
}
```
Can always be replaced by this:
```go
response.rateLimitRemaining, err = strconv.Atoi(limit)
if err != nil {
    return httpResponse{}, fmt.Errorf("X-RateLimit-Remaining: %s", err)
}
```

Regarding the out of range behavior, clamping to the biggest value encodable in the underlying int, seems quite reasonable to me, and is documented as such.

In any case, we need to keep the focus on the intent and not get lost on hypothetical corner cases.

What is the intent and the effect of returning in case of error?
Returning `httpResponse{}` will give the zero value to field `rateLimitRemaining`, which for ints is 0.
Is this what is wanted in case of error? Or is this expected not to be used because the caller is expected to handle the error and not look at `httpResponse` ?